networkx/algorithms/centrality/tests/test_domirank.py

```diff
@@ -0,0 +1,330 @@
+
+    def test_P3_patience(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {0: -2.3474247455596924, 1: 4.49591064453125, 2: -2.3474247455596924}
+        b, _, converged = nx.domirank(G, patience=5)
+        assert converged
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+
+class TestDomirankDirected:
+    @classmethod
+    def setup_class(cls):
+        G = nx.DiGraph()
+
+        edges = [
+            (1, 2),
+            (1, 3),
+            (2, 4),
+            (3, 2),
+            (3, 5),
+            (4, 2),
+            (4, 5),
+            (4, 6),
+            (5, 6),
+            (5, 7),
+            (5, 8),
+            (6, 8),
+            (7, 1),
+            (7, 5),
+            (7, 8),
+            (8, 6),
+            (8, 7),
+        ]
+
+        G.add_edges_from(edges)
+        cls.G = G.reverse()
+        cls.drc = {
+            1: 0.8088147044181824,
+            2: 1.3506824970245361,
+            3: 0.11302878707647324,
+            4: -0.20726482570171356,
+            5: 2.0464682579040527,
+            6: 0.3447999954223633,
+            7: -0.36868923902511597,
+            8: 0.5775589942932129,
+        }
+
+        cls.drca = {
+            1: 0.8100015022345155,
+            2: 1.3502259832012713,
+            3: 0.11228691263288304,
+            4: -0.20697950162756873,
+            5: 2.0479405728029034,
+            6: 0.3427231477861121,
+            7: -0.3705877648939345,
+            8: 0.5791237659444093,
+        }
+
+    def test_domirank_centrality_directed(self):
+        G = self.G
+        b, _, converged = nx.domirank(G)
+        assert converged
+        for n in sorted(G):
+            assert b[n] == pytest.approx(self.drc[n], abs=5e-3)
+
+    def test_domirank_centrality_directed_analytical(self):
+        G = self.G
+        b, _, converged = nx.domirank(G, comp_method="analytical")
+        assert converged == None
+        for n in sorted(G):
+            assert b[n] == pytest.approx(self.drca[n], abs=5e-3)
+
+
+class TestDomirankGraphExceptions:
+    @pytest.mark.parametrize(
+        (
+            "G",
+            "expectation",
+        ),
+        [
+            (
+                nx.Graph(),
+                pytest.raises(nx.NetworkXPointlessConcept),
+            ),
+            (
+                nx.MultiGraph(),
+                pytest.raises(nx.NetworkXNotImplemented),
+            ),
+        ],
+    )
+    def test_domirank_exceptions(self, G, expectation):
+        with expectation:
+            nx.domirank(
+                G,
+            )
+
+
+class TestDomirankMethodExceptions:
+    @pytest.mark.parametrize(
+        (
+            "comp_method",
+            "expectation",
+        ),
+        [
+            (
+                "ThisShouldNotWork",
+                pytest.raises(nx.NetworkXUnfeasible),
+            ),
+        ],
+    )
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        (
            "method",
            "expectation",
        ),
        [
            (
                "ThisShouldNotWork",
                pytest.raises(nx.NetworkXUnfeasible),
            ),
            (
                "iterative",
                does_not_raise(),
            ),
        ],
    )
```
This requires `from contextlib import nullcontext as does_not_raise()` at the top of the file!

This is a general example for the scenario I discuss in the main review comment. Let me know what you think.